bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -280,10 +286,7 @@ public Iterable<HistoricItem> query(FilterCriteria filter) {
         }
 
         long start = 0L;
-        // set end to {@link Instant#MAX} instead of current timestamp to enable requesting future time ranges including
-        // boundary values via REST API
-        // see discussion in https://github.com/openhab/openhab-addons/pull/14238
-        long end = filter.getEndDate() == null ? Instant.MAX.getEpochSecond()
+        long end = filter.getEndDate() == null ? System.currentTimeMillis() / 1000
```

Comment:
Now that we have a local variable, let's use it:
```suggestion
        long end = filterEndDate == null ? System.currentTimeMillis() / 1000
```

also in next line (and possibly elsewhere) - and similarly for `filter.getBeginDate()`.